src/som/compiler/MethodBuilder.java

```diff
@@ -360,22 +354,18 @@ public boolean isBlockMethod() {
   }
 
   private int getOuterSelfContextLevel() {
-    int level = 0;
-    MethodBuilder ctx = outerBuilder;
-    while (ctx != null) {
-      ctx = ctx.outerBuilder;
-      level++;
-    }
-    return level;
+    return countLevelsToNextMixin();
   }
 
   private int getContextLevel(final String varName) {
     if (locals.containsKey(varName) || arguments.containsKey(varName)) {
       return 0;
     }
 
-    if (outerBuilder != null) {
-      return 1 + outerBuilder.getContextLevel(varName);
+    MethodBuilder nextMethod = getNextMethod();
+    if (nextMethod != null) {
+      int nLevels = countLevelsToNextMethod();
```

Comment:
I'd drop the `n` on the name. I am not a fan of 1-char abbreviations, except when I use them myself because of laziness ;)